graphdatascience/query_runner/arrow_query_runner.py

```diff
@@ -81,18 +86,25 @@ def run_query_with_logging(self, query: str, params: Optional[Dict[str, Any]] =
         # For now there's no logging support with Arrow queries.
         if params is None:
             params = {}
+
         return self._fallback_query_runner.run_query_with_logging(query, params)
 
     def set_database(self, db: str) -> None:
         self._fallback_query_runner.set_database(db)
 
-    def database(self) -> str:
+    def database(self) -> Optional[str]:
         return self._fallback_query_runner.database()
 
     def close(self) -> None:
         self._fallback_query_runner.close()
 
     def _run_arrow_property_get(self, graph_name: str, procedure_name: str, configuration: Dict[str, Any]) -> DataFrame:
+        if not self.database():
+            raise ValueError(
+                "For this call you must have explicitly specified a valid Neo4j database to target, "
```

Comment:
maybe `... to execute on` instead of `.. to target`?

Reply:
Sure!